### DISCUSSION TITLE
Do we need a CookieManager embedded in Hydro?

### DISCUSSION THREAD

**Discussion by taublast** [hydrostack/hydro#68] [General]
We could add an implementation of such, already dogfooding this:

```csharp
public interface ICookieManager
{
    public T Get<T>(string key, bool secure, T defaultValue);

    public void Set<T>(string key, T value, bool secure, TimeSpan? expires = null);
}
```

Works well from components code-behind, would this be of interest?

**Comment by kjeske** [comment]
In Hydro component you can already do something similar, but with some more code:

```c#
public void SetCookie()
{
    var cookieValue = HttpContext.Request.Cookies["key"];
    HttpContext.Response.Cookies.Append("key", "value", new CookieOptions { Secure = true });
}
```

So the solution would be probably to only add some extension methods to change it to:
```c#
public void SetCookie()
{
    var cookieValue = this.GetCookie("key");
    this.SetCookie("key", "value", new CookieOptions { Secure = true });
}
```

Not sure yet if it's worth it though, is it?

**Comment by taublast** [comment]
Since we already have newtonsoft Json reference the addition would come at no extra-weight for Hydro, while providing a platform agnostic way to interact wth cookies, simple considering it as some abstract storage. So if we can store value types others than string and entire classes.

Isn't it cool to just write`var lastUsedCar = _storage.Get<Car>("lastCar");` out-of-the-box?

**Reply by kjeske** [reply]
Yes, true, when we talk about serialized object, then there is some logic to handle. [I did that](https://github.com/hydrostack/hydro-sales/blob/main/Utils/CookieStorageProvider.cs) also in the Hydro Sales project, but with fixed storage object and encrypted. IMO it's good to encrypt the content of a cookie value in this case.

So yes, I think it is a nice candidate for a feature :)

**Comment by kjeske** [comment]
I created an issue for that #72 and we can track it there, so closing this one. Thanks for again a nice suggestion!
